### PR TITLE
[294] configured vscode config for indentation, added tslint-eslint-rules

### LINKED ISSUE #294: make sure indentation is done automatically when coding in vscode

### PR DESCRIPTION
**Pull request by adietish**
fixes #294

### CODE REVIEW

**Comment by adietish** [comment]
@dgolovin @jrichter1 dare to review please?

File: .vscode/settings.json [hunk 1]
```diff
@@ -5,5 +5,10 @@
...
         "out": true // set this to false to include "out" folder in search results
-    }
-}
\ No newline at end of file
+    },
+    "[typescript]": {
+      "editor.tabSize": 4,
+      "editor.insertSpaces": true,
+      "editor.detectIndentation": false
+  }
```
Comment (jrichter1) [inline .vscode/settings.json]: This brace is misaligned - transgression!! :smiling_imp:

File: tslint.json [hunk 2]
```diff
@@ -20,15 +25,18 @@
...
             1
         ],
+        "no-console": false,
         "no-default-export": true,
-		"no-duplicate-variable": true,
-		"no-duplicate-imports": true,
-		"no-unused-variable": true,
+		    "no-duplicate-variable": true,
+		    "no-duplicate-imports": true,
+		    "no-unused-variable": true,
```
Comment (jrichter1) [inline tslint.json]: There is no reason these 3 lines should have different indentation to the line above them. Can you fix that?

File: .vscode/settings.json (hunk 1 above)
Comment (adietish) [reply .vscode/settings.json]: response is classified ;)

File: tslint.json (hunk 2 above)
Comment (adietish) [reply tslint.json]: looks like this is github displaying it false. Look how things look in vscode:
image

Comment (jrichter1) [reply tslint.json]: Shows just like on github in my vscode. And the interesting part is that the neighbouring lines are indented by 8 spaces. And these 3 have 2 tabs and 4 spaces. So I guess you have your vscode show tabs as 2 spaces.

Comment (adietish) [reply tslint.json]: @jrichter1 indeed, those 3 had tabs. But then I have vscode set to 4 spaces:
```
 "[typescript]": {
      "editor.tabSize": 4,
```

Comment (adietish) [reply tslint.json]: weird enough, tslint config even has the rule that would enforce spaces and fix it. But it didn't seem to fix the above:
```
        "indent": [
            true,
            "spaces"
        ],
```

**Pull request merged**